romancal/tweakreg/tweakreg_step.py

```diff
@@ -0,0 +1,560 @@
+"""
+Roman pipeline step for image alignment.
+
+:Authors: Mihai Cara & Mairan Teodoro
+
+"""
+from os import path
+
+from astropy.table import Table
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+from tweakwcs.imalign import align_wcs
+from tweakwcs.correctors import JWSTWCSCorrector
+from tweakwcs.matchutils import XYXYMatch
+
+# LOCAL
+from ..stpipe import RomanStep
+from roman_datamodels import datamodels
+from . import astrometric_utils as amutils
+from roman_datamodels.util import is_association
+
+
+def _oxford_or_str_join(str_list):
+    nelem = len(str_list)
+    if not nelem:
+        return 'N/A'
+    str_list = list(map(repr, str_list))
+    if nelem == 1:
+        return str_list
+    elif nelem == 2:
+        return str_list[0] + ' or ' + str_list[1]
+    else:
+        return ', '.join(map(repr, str_list[:-1])) + ', or ' + repr(str_list[-1])
+
+
+SINGLE_GROUP_REFCAT = ['GAIADR3', 'GAIADR2', 'GAIADR1']
+_SINGLE_GROUP_REFCAT_STR = _oxford_or_str_join(SINGLE_GROUP_REFCAT)
+DEFAULT_ABS_REFCAT = SINGLE_GROUP_REFCAT[0]
+ALIGN_TO_ABS_REFCAT = True
+
+__all__ = ['TweakRegStep']
+
+
+class TweakRegStep(RomanStep):
+    """
+    TweakRegStep: Image alignment based on catalogs of sources detected in
+    input images.
+    """
+
+    class_alias = "tweakreg"
+
+    spec = f"""
+        save_catalogs = boolean(default=False) # Write out catalogs?
+        use_custom_catalogs = boolean(default=False) # Use custom user-provided catalogs?
+        catalog_format = string(default='ecsv') # Catalog output file format
+        catfile = string(default='') # Name of the file with a list of custom user-provided catalogs
+        enforce_user_order = boolean(default=False) # Align images in user specified order?
+        expand_refcat = boolean(default=False) # Expand reference catalog with new sources?
+        minobj = integer(default=15) # Minimum number of objects acceptable for matching
+        searchrad = float(default=2.0) # The search radius in arcsec for a match
+        use2dhist = boolean(default=True) # Use 2d histogram to find initial offset?
+        separation = float(default=1.0) # Minimum object separation in arcsec
+        tolerance = float(default=0.7) # Matching tolerance for xyxymatch in arcsec
+        xoffset = float(default=0.0), # Initial guess for X offset in arcsec
+        yoffset = float(default=0.0) # Initial guess for Y offset in arcsec
```

Comment:
I suggest we remove `xoffset` and `yoffset` from the interface for Roman. These are defined in the V2V3 tangent plane and hard for a user to compute. Make sure the defaults are defined in the code. @mcara may confirm this and whether the defaults need to be set in the romancal code

Reply:
I removed `xoffset` and `yoffset` from the `spec` and set both `xoffset` and `yoffset` to zero in the call to `XYXYMatch`.

Reply:
I concur.